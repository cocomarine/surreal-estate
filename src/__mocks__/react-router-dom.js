const reactRouterDom = jest.createMockFromModule("react-router-dom");

reactRouterDom.withRouter = ({ children }) => children;

module.exports = reactRouterDom;
