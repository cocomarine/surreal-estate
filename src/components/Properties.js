import axios from "axios";
import React, { useState, useEffect } from "react";
import { useLocation } from "react-router-dom";
import PropTypes from "prop-types";
import PropertyCard from "./PropertyCard";
import Alert from "./Alert";
import SideBar from "./SideBar";
import "../styles/properties.css";

const Properties = ({ userID }) => {
  const initialState = {
    properties: [],
  };

  const [properties, setProperties] = useState(initialState.properties);
  const [alert, setAlert] = useState({ message: "" });

  useEffect(() => {
    axios
      .get("http://localhost:4000/api/v1/PropertyListing/")
      .then((res) => {
        setProperties(res.data);
        setAlert({
          message: "",
        });
      })
      .catch(() => {
        setAlert({
          message: "Server error. Please try again later.",
        });
      });
  }, []);

  const { search } = useLocation();

  useEffect(() => {
    axios
      .get(`http://localhost:4000/api/v1/PropertyListing${search}`)
      .then((res) => {
        setProperties(res.data);
      })
      .catch((err) => console.error(err));
  }, [search]);

  return (
    <div className="properties-container">
      <div className="sidebar">
        <SideBar />
      </div>
      <div className="properties-cards">
        <Alert message={alert.message} success={alert.isSuccess} />
        {properties.map((property) => (
          <div className="item" key={property._id}>
            <PropertyCard userID={userID} {...property} />
          </div>
        ))}
      </div>
    </div>
  );
};

Properties.propTypes = {
  userID: PropTypes.string.isRequired,
};

export default Properties;
