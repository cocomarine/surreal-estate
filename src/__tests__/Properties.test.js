import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Properties from "../components/Properties";

describe("Properties", () => {
  const testUserID = "test userID";

  it("renders correctly", () => {
    const rendered = renderer.create(
      <MemoryRouter>
        <Properties userID={testUserID} />
      </MemoryRouter>
    );
    expect(rendered).toMatchSnapshot();
  });
});
