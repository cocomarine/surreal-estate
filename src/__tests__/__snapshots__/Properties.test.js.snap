// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties renders correctly 1`] = `
<div
  className="properties-container"
>
  <div
    className="sidebar-container"
  >
    <div
      className="sidebar"
    >
      <form
        className="search-form"
        onSubmit={[Function]}
      >
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search by Title"
          type="text"
        />
        <button
          className="search-button"
          type="submit"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-magnifying-glass magglassIcon"
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </form>
      <div
        className="city-link-container"
      >
        <b>
          Filter by city
        </b>
        <div
          className="city-name"
        >
          <a
            className="city-link"
            href="/?query={\\"city\\":\\"Manchester\\"}"
            onClick={[Function]}
          >
            Manchester
          </a>
        </div>
        <div
          className="city-name"
        >
          <a
            className="city-link"
            href="/?query={\\"city\\":\\"Leeds\\"}"
            onClick={[Function]}
          >
            Leeds
          </a>
        </div>
        <div
          className="city-name"
        >
          <a
            className="city-link"
            href="/?query={\\"city\\":\\"Sheffield\\"}"
            onClick={[Function]}
          >
            Sheffield
          </a>
        </div>
        <div
          className="city-name"
        >
          <a
            className="city-link"
            href="/?query={\\"city\\":\\"Liverpool\\"}"
            onClick={[Function]}
          >
            Liverpool
          </a>
        </div>
        <div
          className="city-name"
        >
          <a
            className="city-link"
            href="/?query={\\"city\\":\\"Birmingham\\"}"
            onClick={[Function]}
          >
            Birmingham
          </a>
        </div>
        <div
          className="city-name"
        >
          <a
            className="city-link"
            href="/?query={\\"city\\":\\"London\\"}"
            onClick={[Function]}
          >
            London
          </a>
        </div>
      </div>
      <div
        className="sort-container"
      >
        <b>
          Sort by
        </b>
        <div
          className="price-sort"
        >
          <a
            className="price-link"
            href="/?sort={\\"price\\":1}"
            onClick={[Function]}
          >
            Price Ascending
          </a>
        </div>
        <div
          className="price-sort"
        >
          <a
            className="price-link"
            href="/?sort={\\"price\\":-1}"
            onClick={[Function]}
          >
            Price Descending
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="properties-cards"
  />
</div>
`;
